files/image_config/monit/process_checker

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python
+import argparse
+import sys
+import syslog
+
+import psutil
+import swsssdk
+
+
+def check_process_existence(container_name, process_name, process_cmdline):
+    """
+    @summary: Check whether the process in the specified container is running or not and
+              an alerting message will written into syslog if it failed to run.
+    """
+    config_db = swsssdk.ConfigDBConnector()
+    config_db.connect()
+    feature_table = config_db.get_table("FEATURE")
+
+    if container_name in feature_table.keys():
+        # We look into the 'FEATURE' table to verify whether the container is disabled or not.
+        # If the container is diabled, we exit.
+        if ("state" in feature_table[container_name].keys()
+                and feature_table[container_name]["state"] == "disabled"):
+            sys.exit(0)
+        else:
+            # We leveraged the psutil library to help us check whether the process is running or not.
+            # If the process entity is found in process tree and it is also in the 'running' or 'sleeping'
+            # state, then it will be marked as 'running'.
+            is_running = False
+            for process in psutil.process_iter(["name", "cmdline", "status"]):
+                # The script process_checker has the command line format '/usr/bin/process_checker <container_name>
+                # <process_name> <process_cmdline>' such as '/usr/bin/process_checker bgp fpmsyncd fpmsyncd'. So
+                # when using psutil to search process 'fpmsyncd', we should skip the process which ran process_checker
+                # since it is not 'fpmsyncd' process although 'fpmsyncd' is a sustring of its cmdline.
+                if process.name() == "process_checker":
+                    continue
```

Comment:
Is this check necessary?

Reply:
Since we already used the `startswith(...)` method to check whether the process command line starts with the string, this check will be unnecessary.